files/image_config/reboot-cause/process-reboot-cause

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+#
+# process-reboot-cause
+#
+# Program designed to read the previous reboot-cause files, log the last previous reboot-cause.
+# And read the saved reboot-cause history files and save the reboot cause in the state-db.
+#
+
+try:
+    import json
+    import os
+    import pwd
+    import re
+    import sys
+
+    import swsssdk
+    from sonic_py_common import device_info, logger
+except ImportError as err:
+    raise ImportError("%s - required module not found" % str(err))
+
+VERSION = "1.0"
+
+SYSLOG_IDENTIFIER = "process-reboot-cause"
+
+REBOOT_CAUSE_DIR = "/host/reboot-cause/"
+REBOOT_CAUSE_HISTORY_DIR = "/host/reboot-cause/history/"
+PREVIOUS_REBOOT_CAUSE_FILE = REBOOT_CAUSE_DIR + "previous-reboot-cause.json"
+USER_ISSUE_REBOOT_CAUSE_REGEX ="User issued \'{}\' command [User: {}, Time: {}]"
+
+REBOOT_CAUSE_UNKNOWN = "Unknown"
+REBOOT_CAUSE_TABLE_NAME = "REBOOT_CAUSE"
+
+REDIS_HOSTIP = "127.0.0.1"
+state_db = None
+
+# Global logger class instance
+sonic_logger = logger.Logger(SYSLOG_IDENTIFIER)
+
+
+# ============================= Functions =============================
+def set_reboot_cause_table_entry(key, cause, time, user, comment):
+    # Connect State DB
+    state_db = swsssdk.SonicV2Connector(host=REDIS_HOSTIP)
+    state_db.connect(state_db.STATE_DB)
+
+    state_db.set(state_db.STATE_DB, key, 'cause', cause)
+    state_db.set(state_db.STATE_DB, key, 'time', time)
+    state_db.set(state_db.STATE_DB, key, 'user', user)
+    state_db.set(state_db.STATE_DB, key, 'comment', comment)
+
+
+def read_reboot_cause_files_and_save_state_db():
+    # Sort the previous reboot cause files by creation time
+    FULL_REBOOT_FILE_LIST = [os.path.join(REBOOT_CAUSE_HISTORY_DIR,i) for i in os.listdir(REBOOT_CAUSE_HISTORY_DIR)]
+    TIME_SORTED_FULL_REBOOT_FILE_LIST = sorted(FULL_REBOOT_FILE_LIST, key=os.path.getmtime, reverse=True)
```

Comment:
Suggest removing "FULL" from constant names. Seems unnecessary, as there are no references to a other/partial lists.

Reply:
Also remove "FULL from "TIME_SORTED_FULL_REBOOT_FILE_LIST"